client/my-sites/plugins/plugins-list/index.tsx

```diff
@@ -178,17 +215,17 @@ export class PluginsList extends Component {
 			}
 			return false;
 		},
-		updates( plugin ) {
+		updates: ( plugin: Plugin ) => {
 			if ( this.isSelected( plugin ) ) {
 				return Object.keys( plugin.sites ).some( ( siteId ) => {
 					const sitePlugin = getSitePlugin( plugin, siteId, this.props.pluginsOnSites );
-					const site = this.props.allSites.find( ( s ) => s.ID === parseInt( siteId ) );
-					return sitePlugin?.update?.new_version && site.canUpdateFiles;
+					const site = this.props.allSites.find( ( s ) => s && s.ID === parseInt( siteId ) );
+					return site && sitePlugin?.update?.new_version && site.canUpdateFiles;
```

Comment:
Nit: We could just use optional chaining to make it shorter:

```suggestion
					return sitePlugin?.update?.new_version && site?.canUpdateFiles;
```